client_plugin/drivers/shared/shared_driver.go

```diff
@@ -145,7 +147,8 @@ func NewVolumeDriver(cfg config.Config, mountDir string) *VolumeDriver {
 	d.dockerd = cli
 
 	// initialize built-in etcd cluster
-	err = d.initEtcd()
+	d.etcd.driver = &d
```

Comment:
No really :)
For example, from `etcdEventHandler` function (which is a method of `etcdInfo` struct), we need to call `startSMBServer` function (which is a method of `VolumeDriver` struct). So from `e *etcdInfo` we need to get the VolumeDriver associate with it, and the best way is to link back VolumeDriver inside etcdInfo.